src/VisualStudio/Core/Impl/ProjectSystem/CPS/CPSProject_IWorkspaceProjectContext.cs

```diff
@@ -108,29 +109,14 @@ public new void RemoveMetadataReference(string referencePath)
         public void AddProjectReference(IWorkspaceProjectContext project, MetadataReferenceProperties properties)
         {
             var abstractProject = GetAbstractProject(project);
-
-            // AbstractProject and ProjectTracker track project references using the project bin output path.
-            // Setting the command line arguments should have already set the output file name and folder.
-            // We fetch this output path to add the reference.
-            var referencedProject = this.ProjectTracker.GetProject(abstractProject.Id);
-            var binPathOpt = referencedProject.BinOutputPath;
-            if (!string.IsNullOrEmpty(binPathOpt))
-            {
-                AddMetadataReferenceAndTryConvertingToProjectReferenceIfPossible(binPathOpt, properties);
-            }
+            AddProjectReference(new ProjectReference(abstractProject.Id, properties.Aliases, properties.EmbedInteropTypes));
```

Comment:
It shouldn't be necessary to update that map for things that are "true" project references added via AddProjectReference.  That map is only for things that we have promoted to P2P reference based on the target path.

What user visible problem were you seeing when calling this?